client/state/data-layer/wpcom/jetpack-onboarding/index.js

```diff
@@ -53,6 +97,9 @@ export const announceSaveFailure = ( { dispatch } ) =>
 	dispatch( errorNotice( translate( 'An unexpected error occurred. Please try again later.' ) ) );
 
 export default {
+	[ JETPACK_ONBOARDING_SETTINGS_REQUEST ]: [
+		dispatchRequest( fetchSettings, addSettings, announceFetchFailure ),
```

Comment:
With the current implementation, it probably makes sense to use `dispatchRequestEx` instead. However, in order to make this work for non-connected sites, we'll need to change the implementation to add the token and email to the request, thus will need access to the state.